src/util/segmentation/operations/eraseInside.js

```diff
@@ -0,0 +1,59 @@
+import { getBoundingBoxAroundPolygon } from '../boundaries';
+import pointInPolygon from '../../pointInPolygon';
+
+import { getLogger } from '../../logger';
+
+const logger = getLogger('util:segmentation:operations:eraseInside');
+
+/**
+ * EraseInside - Erase all pixels labeled with the activeSegmentIndex,
+ * in the region defined by evt.operationData.points.
+ * @param  {} evt The Cornerstone event.
+ * @param {} evt.operationData An object containing the `pixelData` to
+ *                          modify, the `segmentIndex` and the `points` array.
+ * @returns {null}
+ */
+export default function eraseInside(evt) {
+  const eventData = evt.detail;
+  const { operationData } = evt;
+  const { pixelData, segmentIndex, points } = operationData;
+
+  if (operationData.segmentationMixinType !== `freehandSegmentationMixin`) {
+    logger.error(
+      `eraseInside operation requires freehandSegmentationMixin operationData, recieved ${
+        operationData.segmentationMixinType
+      }`
+    );
+
+    return;
+  }
```

Comment:
Do we need this check? Why do we care what the mixin's type is? So long as the arguments are satisfied for the method call, shouldn't that be all that matters?

Reply:
The `operationData` can be anything based on what the mixin does. If the mixin makes you place two control points, `eraseInside` isn't an appropriate solution, so we need to check they are compatible and warn if users build tools that don't make sense.

Reply:
Talked about this, will clean up as a larger refactor of the segmentation utils near the end.